docs/tutorial/resift-rentals.md

```diff
@@ -0,0 +1,2110 @@
+
+```js
+// App.js
+import { Guard } from 'resift';
+```
+
+And wrap the map function in `Guard`
+
+```js
+<Guard fetch={getGenres}>
+  {genres => genres.map(genre => <Genre key={genre.id} genre={genre} className={classes.genre} />)}
+</Guard>
+```
+
+If you refresh the page now, we'll the error is gone, but the genres data is still not showing up on the page. Why is that?
+The reason is because the genres will always be null on the first render because we have not dispatched the fetch. This is when `useDispatch` comes in.
+
+#### Step 4: Dispatch the Fetch
+
+Data dispatch should happen in one of the two occasions: 1) when a page first loads/when the component mounts; 2) when there's an event kicks in defined by the event handler. Our case is the former, we need the genres data when we load the page. For this we'll use React's [useEffect hook](https://reactjs.org/docs/hooks-effect.html). Let's add the imports and the effect:
+
+```js
+// App.js
+...
+import { useEffect } from 'react';
+import { useDispatch } from 'resift'
+
+function App() {
+  ...
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getGenres());
+  }, [dispatch]);
+}
+```
+
+The complete code at this point looks like this:
+
+```js
+// App.js
+import React, { useEffect } from 'react';
+// Components
+import AppBar from 'components/AppBar';
+import Genre from 'components/Genre';
+// Fetches
+import { useData, useDispatch, Guard } from 'resift';
+import getGenres from 'fetches/getGenres';
+// Styles
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles(theme => ({
+  root: {},
+  genre: {
+    margin: '24px 0',
+  },
+  spinner: {
+    color: 'white',
+  },
+}));
+
+function App() {
+  const classes = useStyles();
+  const genres = useData(getGenres);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getGenres());
+  }, [dispatch]);
+
+  return (
+    <div className={classes.root}>
+      <AppBar />
+      <Guard fetch={getGenres}>
+        {genres => genres.map(genre => <Genre key={genre.id} genre={genre} />)}
+      </Guard>
+    </div>
+  );
+}
+
+export default App;
+```
+
+Refresh the page now and you'll see the genres data load after a second. Wait, that's not a great user experience having to wait for the data to come without know what's happening. We should indicate to our user that the data is loading.
+
+![Waiting GIF](https://media.giphy.com/media/9SIXFu7bIUYHhFc19G/giphy.gif)
+
+#### Show Fetch Status
+
+We want to show a loading spinner when we're fetching the data. To achieve this, we'll use the `useStatus` and a helper function to check is the status is 'loading' called `isLoading`. And we'll grab the Material UI spinner called `CircularProgress`. Let's get them imported.
+
+```js
+import { useStatus, isLoading } from 'resift';
+import { CircularProgress } from '@material-ui/core';
+```
+
+Now we can add the spinner in while loading
+
+```js
+function App() {
+  ...
```

Comment:
This is very very nit-picky so feel free to ignore this stylistic comment:

I think it would be better if the `...` here was inside a comment. e.g.

```suggestion
  // ...
```

Because it's not in a comment, this code block is not syntactically correct javascript and thus prettier can't format this block for you ([prettier formats code inside markdown files](https://prettier.io/blog/2017/11/07/1.8.0.html#support-markdown-2943httpsgithubcomprettierprettierpull2943-by-ikatyanghttpsgithubcomikatyang)). If you choose to work on this, you can probably find and replace all the `...`s.